subsys/net/lib/sockets/sockets.c

```diff
@@ -465,7 +465,10 @@ int zsock_accept_ctx(struct net_context *parent, struct sockaddr *addr,
 	ctx = k_fifo_get(&parent->accept_q, timeout);
 	if (ctx == NULL) {
 		z_free_fd(fd);
-		errno = EAGAIN;
+		/* Return EINVAL which is the same error code used under Linux
+		 * when calling shutdown on a blocked accept call
+		 */
+		errno = EINVAL;
```

Comment:
Hmm, but if the timeout was K_NO_WAIT, then the EAGAIN would be appropriate here.

Reply:
Good point, I added a condition